admin/win/msi/Nextcloud.wxs

```diff
@@ -0,0 +1,207 @@
+        Id="*" 
+        UpgradeCode="$(var.UpgradeCode)"
+        Language="1033" Codepage="$(var.codepage)" Version="$(var.VerFull)">
+    <Package Id="*" Keywords="Installer" Description="$(var.AppName) $(var.VerDesc)" Manufacturer="$(var.AppVendor)"
+        InstallerVersion="300" Platform="$(var.Platform)" Languages="1033" Compressed="yes" SummaryCodepage="$(var.codepage)" InstallScope="perMachine" />
+
+    <!--
+        Upgrading: Since we always want to allow up-/downgrade, we don't specify a maximum version, thus
+                   leading the WiX linker (light.exe) to trigger the following warning:
+                        warning LGHT1076 : ICE61: This product should remove only older versions of itself. No Maximum version was detected for the current product. (WIX_UPGRADE_DETECTED)
+                   We suppress the warning: light.exe -sw1076
+        
+        If at some point we want to change this behaviour, read the docs:
+        https://www.firegiant.com/wix/tutorial/upgrades-and-modularization/replacing-ourselves/
+        https://www.joyofsetup.com/2010/01/16/major-upgrades-now-easier-than-ever/
+    -->
+    <MajorUpgrade Schedule="afterInstallInitialize" AllowDowngrades="yes" />
+
+    <Media Id="1" Cabinet="$(var.AppShortName).cab" EmbedCab="yes" />
+
+    <!-- If already installed: Use previously chosen path (use 32-bit registry like NSIS does) -->
+    <Property Id="INSTALLDIR">
+        <RegistrySearch Id="RegistryInstallDir" Type="raw" Root="HKLM" Key="Software\$(var.AppVendor)\$(var.AppName)" Win64="no" />
+    </Property>
+
+    <!-- Detect legacy NSIS installation -->
+    <Property Id="NSIS_UNINSTALLEXE">
+        <DirectorySearch Id="LegacyUninstallVersion" Path="[INSTALLDIR]">
+            <FileSearch Name="Uninstall.exe" />
+        </DirectorySearch>
+    </Property>
+
+    <!-- Quit / restart application -->
+    <util:RestartResource ProcessName="$(var.AppExe)" />
+
+    <!-- Helper DLL Custom Actions -->
+    <SetProperty Id="ExecNsisUninstaller" Value="&quot;$(var.AppShortName)&quot; &quot;[NSIS_UNINSTALLEXE]&quot;" Before="ExecNsisUninstaller" Sequence="execute" />
+    <SetProperty Id="RemoveNavigationPaneEntries" Value="&quot;$(var.AppName)&quot;" Before="RemoveNavigationPaneEntries" Sequence="execute" />
+
+    <InstallExecuteSequence>
+        <!-- Install: Remove previous NSIS installation, if detected -->
+        <Custom Action="ExecNsisUninstaller" Before="ProcessComponents">NSIS_UNINSTALLEXE AND NOT Installed</Custom>
+
+        <!-- Uninstall: Remove sync folders from Explorer's Navigation Pane, only effective for the current user (home users) -->
+        <Custom Action="RemoveNavigationPaneEntries" After="RemoveFiles">(NOT UPGRADINGPRODUCTCODE) AND (REMOVE="ALL")</Custom>
+
+        <!-- Schedule Reboot for the Shell Extensions -->
+        <ScheduleReboot After="InstallFinalize">NOT (DO_NOT_SCHEDULE_REBOOT=1)</ScheduleReboot>
```

Comment:
Yeah this MSI conditions are really odd and counter-intuitive in general, you would have fun xD

Reply:
In this case it is to ensure only to _NOT_ ask for a reboot if `DO_NOT_SCHEDULE_REBOOT=1` ;-)

Reply:
Yeah but then would it be like `DO_NOT_SCHEDULE_REBOOT>0`? That's what I kind of found odd, testing against a value and then negating the result.

Reply:
This is the confusing thing in the beginning: MSI conditions are the negation of logic, here you test what's not be true ^^
Gets clearer when you look at the feature section at the end of the file, how relative to the element its conditions have do be defined. It's really odd, actually not a WiX problem because they have to respect the way Microsoft designed the MSI thing.
My decision to keep it that way is based on being explicit and trying to be consistent with the other `=1` conditions in the file, even though I don't love it. But I like it more than `>0` ;p